x/batch/types/msg_batch.go

```diff
@@ -0,0 +1,92 @@
+package types
+
+import (
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/axelarnetwork/utils/funcs"
+	"github.com/axelarnetwork/utils/slices"
+)
+
+var _ cdctypes.UnpackInterfacesMessage = BatchRequest{}
+
+// NewBatchRequest is the constructor for BatchRequest
+func NewBatchRequest(sender sdk.AccAddress, messages []sdk.Msg) *BatchRequest {
+	return &BatchRequest{
+		Sender:   sender,
+		Messages: slices.Map(messages, func(msg sdk.Msg) cdctypes.Any { return *funcs.Must(cdctypes.NewAnyWithValue(msg)) }),
+	}
+}
+
+// Route returns the route for this message
+func (m BatchRequest) Route() string {
+	return RouterKey
+}
+
+// Type returns the type of the message
+func (m BatchRequest) Type() string {
+	return "Batch"
+}
```

Comment:
I would prefer if we don't even implement the legacy msg functions anymore if it's not necessary